opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -1,60 +1,122 @@
-use opentelemetry::logs::{LogRecord, Logger, LoggerProvider, Severity};
+use opentelemetry::{
+    logs::{AnyValue, LogRecord, Logger, LoggerProvider, Severity},
+    Key,
+};
 use std::borrow::Cow;
-use tracing_core::Level;
+use tracing_core::{Level, Metadata};
+use tracing_log::NormalizeEvent;
 use tracing_subscriber::Layer;
 
 const INSTRUMENTATION_LIBRARY_NAME: &str = "opentelemetry-appender-tracing";
+// Start by "log."
+const LOG_METADATA_ATTRIBUTES: &[&str] = &[
+    "name",
+    "target",
+    "module.path",
+    "file.path",
+    "file.name",
+    "file.line",
+    "module_path",
+    "file",
+    "line",
+];
 
 /// Visitor to record the fields from the event record.
-struct EventVisitor<'a> {
-    log_record: &'a mut LogRecord,
+#[derive(Default)]
+struct EventVisitor {
+    log_record_attributes: Vec<(Key, AnyValue)>,
+    log_record_body: Option<AnyValue>,
 }
 
-impl<'a> tracing::field::Visit for EventVisitor<'a> {
+fn is_metadata(field: &str) -> bool {
+    if let Some(log_field) = field.strip_prefix("log.") {
+        for log_metadata_field in LOG_METADATA_ATTRIBUTES {
+            if *log_metadata_field == log_field {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+fn get_filename(filepath: &str) -> &str {
+    if let Some((_, filename)) = filepath.rsplit_once('/') {
+        return filename;
+    }
+    if let Some((_, filename)) = filepath.rsplit_once('\\') {
+        return filename;
+    }
+    filepath
+}
+
+impl EventVisitor {
+    fn visit_metadata(&mut self, meta: &Metadata) {
+        self.log_record_attributes
+            .push(("log.name".into(), meta.name().into()));
```

Comment:
Should we remove the redundant attribute here - https://github.com/open-telemetry/opentelemetry-rust/blob/33b5e9b4e2b1ce7b1ed209ad61e2fbcad73e627e/opentelemetry-appender-tracing/src/layer.rs#L109C12-L109C12 ? This will break user_events log exporter, but I can raise a subsequent PR to handle it.

Reply:
Do you talk about this line `let vec = vec![("name", meta.name())];`?

I think yes because when I read this doc https://opentelemetry.io/docs/specs/semconv/general/logs/, it gives me the impression that the best name for this attribute is "log.name". This doc reminds me the Elastic Common Schema. ECS gives you guidelines for custom field names to make you able to give the impression that your custom fields are like a part of the ECS even if it is not.